src/course-outline/data/apiHooks.ts

```diff
@@ -33,3 +33,11 @@ export const useCourseItemData = (itemId?: string, enabled: boolean = true) => (
     enabled: enabled && itemId !== undefined,
   })
 );
+
+export const useCourseDetails = (courseId?: string) => (
+  useQuery({
+    queryKey: courseOutlineQueryKeys.courseDetails(courseId),
+    queryFn: () => getCourseDetails(courseId!),
+    enabled: courseId !== undefined,
```

Comment:
This is the recommended approach for typesafety ([docs](https://tanstack.com/query/latest/docs/framework/react/guides/disabling-queries#typesafe-disabling-of-queries-using-skiptoken)):
```suggestion
    queryFn: courseId ? () => getCourseDetails(courseId) : skipToken,
```

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/2567/commits/3b32d20e1a8736be0743559da2771d5b30e3f29a